src/GovUk.Education.ExploreEducationStatistics.Admin/Services/TopicService.cs

```diff
@@ -163,73 +163,113 @@ public class TopicService : ITopicService
-
-                        _statisticsContext.Release.Remove(statsRelease);
-                        _contentContext.Releases.Remove(contentRelease);
-                    }
-
+                    var publicationIdsToDelete = topic.Publications
+                        .Select(p => p.Id)
+                        .ToList();
+
+                    return await DeleteMethodologiesForPublications(publicationIdsToDelete)
+                        .OnSuccess(() => DeleteReleasesForPublications(publicationIdsToDelete));
+                })
+                .OnSuccess(async topic =>
+                {
+                    _contentContext.Topics.Remove(topic);
                     await _contentContext.SaveChangesAsync();
-                    await _statisticsContext.SaveChangesAsync();
+                    return await _publishingService.TaxonomyChanged();
+                });
+        }
 
-                    _contentContext.Topics.Remove(
-                        await _contentContext
-                            .Topics
-                            .Where(t => t.Id == topic.Id)
-                            .SingleAsync()
-                    );
+        private async Task<Either<ActionResult, Unit>> DeleteMethodologiesForPublications(IEnumerable<Guid> publicationIds)
+        {
+            var methodologyIdsToDelete = await _contentContext
+                .PublicationMethodologies
+                .Where(pm => pm.Owner && publicationIds.Contains(pm.PublicationId))
+                .Select(pm => pm.MethodologyId)
+                .ToListAsync();
+
+            return await methodologyIdsToDelete
+                .Select(methodologyId => _methodologyService.DeleteMethodology(methodologyId, true))
+                .OnSuccessAll()
+                .OnSuccessVoid();
+        }
+
+        private async Task<Either<ActionResult, Unit>> DeleteReleasesForPublications(IEnumerable<Guid> publicationIds)
+        {
+            // Some Content Db Releases may be soft-deleted and therefore not visible.
+            // Ignore the query filter to make sure they are found
+            var releasesIdsToDelete = _contentContext
+                .Releases
+                .IgnoreQueryFilters()
+                .Where(r => publicationIds.Contains(r.PublicationId))
+                .ToList()
+                .OrderBy(release => new IdAndPreviousVersionIdPair(release.Id, release.PreviousVersionId),
+                    VersionedEntityComparer)
+                .Select(r => r.Id);
+
+            return await releasesIdsToDelete.Select(DeleteContentAndStatsRelease)
+                .OnSuccessAll()
+                .OnSuccessVoid();
+        }
 
+        private async Task<Either<ActionResult, Unit>> DeleteContentAndStatsRelease(Guid releaseId)
+        {
+            var contentRelease = await _contentContext
+                .Releases
+                .SingleOrDefaultAsync(r => r.Id == releaseId);
+            
+            if (contentRelease == null)
```

Comment:
We won't be able to do that unless `ReleaseDataFileService` and `ReleaseFileService` are also made aware that they need to ignore filters when force deleting their entities, as the Release won't be found and they will return a `NoContent` result when checking the Release exists. This will propagate back up which means deleting the Topic would ultimately end with a 404.

I already thought about making a change to them to be aware of soft deleted releases too and ignore the filters within them but spotted the ReleaseFiles and Files are deleted when we soft delete the Release anyway so they wouldn't find anything to delete.

The first two operations to delete ReleaseFiles and Files are service calls which can fail so the code makes sure the Releases are not deleted if there's an early failure.

```
If Content Release was not found:

Don't attempt to delete ReleaseFiles and Files as they will fail with NotFound
Delete soft-deleted Release if it's found
Delete Stats Release
Return early.

If Content release was found:

Attempt delete ReleaseFiles
Attempt delete Files
If successful delete both Content and Stats Releases
```